rust/types/src/partitioned_mutex.rs

```diff
@@ -0,0 +1,60 @@
+use std::{
```

Comment:
why is this in the types crate?

Reply:
Maybe cache/ is a better fit or just the crate its used in if only 1

Reply:
We have three uses of it - in the FE dedup, in hnsw load synchronization and in spann posting list synchronization. I thought of putting it in the types crate since it is a new type that at least two crates need

Reply:
Types is more for types - like data models. I think cache/ makes more sense